package/tests/test_server.py

```diff
@@ -175,15 +182,44 @@ def test_no_browser(cli_runner):
     assert server.webbrowser.open_new.called
 
 
+def test_load_file_outside_kedro_project(cli_runner, tmp_path):
+    """
+    Check that running viz with `--load-file` flag works outside of a Kedro project
+    """
+    filepath_json = tmp_path / "test.json"
+    data = {"nodes": None, "edges": None, "tags": None}
+    with open(str(filepath_json), "w") as f:
+        json.dump(data, f)
+
+    result = cli_runner.invoke(
+        server.commands, ["viz", "--load-file", "{}".format(str(filepath_json))]
+    )
+    assert result.exit_code == 0, result.output
+
+
+def test_no_load_file(cli_runner):
+    """
+    Check that running viz without `--load-file` flag should fail outside of a Kedro project
+    """
+    result = cli_runner.invoke(server.commands, ["viz"])
+    assert result.exit_code == 1
+    assert (
+        "Could not find '.kedro.yml'. Make sure that you are inside a Kedro project"
+        in result.output
+    )
+
+
 def test_root_endpoint(client):
     """Test `/` endoint is functional"""
     response = client.get("/")
     assert response.status_code == 200
     assert "Kedro Viz" in response.data.decode()
 
 
-def test_nodes_endpoint(client):
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_nodes_endpoint(cli_runner, client):
     """Test `/api/nodes.json` endoint is functional and returns a valid JSON"""
+    cli_runner.invoke(server.commands, ["viz", "--port", "8000"])
```

Comment:
I added this line as a workaround because this unit test was dependent on other unit tests above (but shouldn't), and started failing when I added new unit tests.